src/server/game/Entities/AreaTrigger/AreaTrigger.h

```diff
@@ -91,6 +92,20 @@ class TC_GAME_API AreaTrigger : public WorldObject, public GridObject<AreaTrigge
         ::Movement::Spline<int32> const& GetSpline() const { return *_spline; }
         uint32 GetElapsedTimeForMovement() const { return GetTimeSinceCreated(); } /// @todo: research the right value, in sniffs both timers are nearly identical
 
+        void InitCircularMovement(AreaTriggerCircularMovementInfo const& cmi, uint32 timeToTarget);
+        AreaTriggerCircularMovementInfo GetAreaTriggerCircularMovementInfo() const { return _areaTriggerCircularMovementInfo; }
+
+        Optional<ObjectGuid> GetCircularMovementCenterGUID() const { return _areaTriggerCircularMovementInfo.TargetGUID; }
+        void SetCircularMovementCenterGUID(ObjectGuid const& guid) { _areaTriggerCircularMovementInfo.TargetGUID = guid; }
+        void ResetCircularMovementCenterGUID() { _areaTriggerCircularMovementInfo.TargetGUID.reset(); }
+
+        Position const* GetCircularMovementCenterPosition() const;
+        void SetCircularMovementCenterPosition(Position const& pos) { _areaTriggerCircularMovementInfo.Center = pos; }
+        void ResetCircularMovementCenterPosition() { _areaTriggerCircularMovementInfo.Center.reset(); }
+
+        // Won't work if its used after the AreaTrigger was created
+        void SetCircularMovementStartDelay(uint32 delay) { _areaTriggerCircularMovementInfo.StartDelay = delay; }
```

Comment:
thats why there is a todo in InitCircularMovement

Reply:
There is nothing that indicates that RE_SHAPE is there for this purpose. For these specific areatriggers which i have sniffs for.. RE_SHAPE is sent immediatley after create. There is no way for me to find out what RE_SHAPE should do AND its named StartDelay for a Reason. Its only active in the first Seconds after creation. You can't delay it anymore after you delayed it.

Reply:
Oh and btw.. for other circular areatriggers, there is no reshape.